An error occurred while compiling clojerl

$ git clone https://github.com/jfacorro/clojerl
Cloning into 'clojerl'...
remote: Counting objects: 8623, done.
remote: Compressing objects: 100% (136/136), done.
Receiving objects:  99% (8553remote: Total 8623 (delta 109), reused 170 (delta 90), pack-r
eused 8377 623)
Receiving objects: 100% (8623/8623), 2.57 MiB | 26.00 KiB/s, done.
Resolving deltas: 100% (6206/6206), done.

$ make
[0;32m===> Verifying dependencies...
[0m[0;32m===> Compiling clojerl
[0mCompiling clojure.core...escript: exception error: undefined function clojerl:start/0
  in function  erl_eval:do_apply/6 (erl_eval.erl, line 670)
  in call from erl_eval:expr/5 (erl_eval.erl, line 438)
  in call from escript:eval_exprs/5 (escript.erl, line 873)
  in call from erl_eval:local_func/6 (erl_eval.erl, line 557)
  in call from escript:interpret/4 (escript.erl, line 789)
  in call from escript:start/1 (escript.erl, line 277)
  in call from init:start_em/1 (init.erl, line 1083)
make[1]: *** [Makefile:77: /clojerl/ebin/clojure.core.beam] Error 127
[0;31m===> [1mHook for compile failed!

[0m[0mmake: *** [Makefile:17: compile] Error 1

@jfacorro 
`
$ make clean; DEBUG=1 make | tee rebar3.debug
[0;32m===> Cleaning out clojerl...
[0m===> Evaluating config script "rebar.config.script"
===> Expanded command sequence to be run: [{default,app_discovery},
                                                  {default,install_deps},
                                                  {default,lock},
                                                  {default,compile}]
===> Evaluating config script "d:/shaman/dev/clojerl/rebar.config.script"
===> Evaluating config script "d:/shaman/dev/clojerl/rebar.config.script"
===> Verifying dependencies...
===> run_hooks("d:/shaman/dev/clojerl", pre_hooks, compile) -> no hooks defined

===> sh info:
        cwd: "d:/shaman/dev/clojerl"
        cmd: cmd /c mklink /j "d:\\shaman\\dev\\clojerl\\_build\\default\\lib\\clojerl\\in
clude" "d:\\shaman\\dev\\clojerl\\include"

===>    opts: [{use_stdout,false},return_on_error]

===> Port Cmd: cmd /q /c cmd /c mklink /j "d:\\shaman\\dev\\clojerl\\_build\\default\\lib\
\clojerl\\include" "d:\\shaman\\dev\\clojerl\\include"
Port Opts: [exit_status,{line,16384},use_stdio,stderr_to_stdout,hide,eof]

===> sh info:
        cwd: "d:/shaman/dev/clojerl"
        cmd: cmd /c mklink /j "d:\\shaman\\dev\\clojerl\\_build\\default\\lib\\clojerl\\sr
c" "d:\\shaman\\dev\\clojerl\\src"

===>    opts: [{use_stdout,false},return_on_error]

===> Port Cmd: cmd /q /c cmd /c mklink /j "d:\\shaman\\dev\\clojerl\\_build\\default\\lib\
\clojerl\\src" "d:\\shaman\\dev\\clojerl\\src"
Port Opts: [exit_status,{line,16384},use_stdio,stderr_to_stdout,hide,eof]

===> Compiling clojerl
===> erlopts [debug_info,debug_info]
===> files to compile ["d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/rbdict.er
l",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IType.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IStringable.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IStack.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.ISet.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.ISequential.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.ISeqable.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.ISeq.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IReduce.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IRecord.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.INamed.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IMeta.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IMap.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.ILookup.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IIndexed.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IHash.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IFn.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IExceptionInfo.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IError.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IEquiv.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IDeref.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.ICounted.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IColl.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IChunkedSeq.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IChunk.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IAssociative.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.Vector.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.Vector.ChunkedSeq.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.TupleMap.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.TupleChunk.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.SortedSet.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.SortedMap.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.Set.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.Range.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.Map.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.List.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.LazySeq.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.Cons.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.ChunkedCons.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Var.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Symbol.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.String.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Reduced.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.reader.TaggedLiteral.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.reader.ReaderConditional.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Nil.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Namespace.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.MultiFn.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Keyword.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Integer.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Float.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.ExceptionInfo.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Default.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Boolean.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.BitString.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Atom.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.util.UUID.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.util.Regex.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.util.Date.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.Type.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.Tuple.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.Reference.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.Process.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.Port.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.Map.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.List.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.io.StringWriter.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.io.StringReader.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.io.PushbackReader.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.io.IWriter.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.io.IReader.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.io.IPushbackReader.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.io.File.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.io.Closeable.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.Fn.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
ojerl_sup.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
ojerl.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_utils.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_scope.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_rt.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_reader.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_protocol.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_murmur3.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_module.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_env.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_emitter_pattern.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_emitter.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_compiler.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_cache.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_analyzer.erl"]
===>      Compiled erlang.io.IPushbackReader.erl
===>      Compiled clojerl.ISeq.erl
===>      Compiled clojerl.ISet.erl
===>      Compiled erlang.io.IWriter.erl
===>      Compiled erlang.io.IReader.erl
===>      Compiled clojerl.IExceptionInfo.erl
===>      Compiled clojerl.IColl.erl
===>      Compiled clojerl.IHash.erl
===>      Compiled clojerl.IMap.erl
===>      Compiled clojerl.IIndexed.erl
===>      Compiled clojerl.IChunkedSeq.erl
===>      Compiled clojerl.IDeref.erl
===>      Compiled clojerl.INamed.erl
===>      Compiled clojerl.IChunk.erl
===>      Compiled clojerl.ISequential.erl
===>      Compiled clojerl.IAssociative.erl
===>      Compiled clojerl.IStringable.erl
===>      Compiled clojerl.IError.erl
===>      Compiled clojerl.IFn.erl
===>      Compiled clojerl.IMeta.erl
===>      Compiled erlang.io.Closeable.erl
===>      Compiled clojerl.IReduce.erl
===>      Compiled clojerl.IEquiv.erl
===>      Compiled clojerl.ISeqable.erl
===>      Compiled clojerl.ILookup.erl
===>      Compiled clojerl.IStack.erl
===>      Compiled clojerl.ICounted.erl
===>      Compiled rbdict.erl
===>      Compiled clojerl.IType.erl
===>      Compiled clojerl.IRecord.erl
===>      Compiled clojerl.Vector.erl
===>      Compiled clojerl.Vector.ChunkedSeq.erl
===>      Compiled clojerl.TupleMap.erl
===>      Compiled clojerl.TupleChunk.erl
===>      Compiled clojerl.SortedSet.erl
===>      Compiled clojerl.SortedMap.erl
===>      Compiled clojerl.Set.erl
===>      Compiled clojerl.Range.erl
===>      Compiled clojerl.Map.erl
===>      Compiled clojerl.List.erl
===>      Compiled clojerl.LazySeq.erl
===>      Compiled clojerl.Cons.erl
===>      Compiled clojerl.ChunkedCons.erl
===>      Compiled clojerl.Var.erl
===>      Compiled clojerl.Symbol.erl
===>      Compiled clojerl.String.erl
===>      Compiled clojerl.Reduced.erl
===>      Compiled clojerl.reader.TaggedLiteral.erl
===>      Compiled clojerl.reader.ReaderConditional.erl
===>      Compiled clojerl.Nil.erl
===>      Compiled clojerl.Namespace.erl
===>      Compiled clojerl.MultiFn.erl
===>      Compiled clojerl.Keyword.erl
===>      Compiled clojerl.Integer.erl
===>      Compiled clojerl.Float.erl
===>      Compiled clojerl.ExceptionInfo.erl
===>      Compiled clojerl.Default.erl
===>      Compiled clojerl.Boolean.erl
===>      Compiled clojerl.BitString.erl
===>      Compiled clojerl.Atom.erl
===>      Compiled erlang.util.UUID.erl
===>      Compiled erlang.util.Regex.erl
===>      Compiled erlang.util.Date.erl
===>      Compiled erlang.Type.erl
===>      Compiled erlang.Tuple.erl
===>      Compiled erlang.Reference.erl
===>      Compiled erlang.Process.erl
===>      Compiled erlang.Port.erl
===>      Compiled erlang.Map.erl
===>      Compiled erlang.List.erl
===>      Compiled erlang.io.StringWriter.erl
===>      Compiled erlang.io.StringReader.erl
===>      Compiled erlang.io.PushbackReader.erl
===>      Compiled erlang.io.File.erl
===>      Compiled erlang.Fn.erl
===>      Compiled clojerl_sup.erl
===>      Compiled clojerl.erl
===>      Compiled clj_utils.erl
===>      Compiled clj_scope.erl
===>      Compiled clj_rt.erl
===>      Compiled clj_reader.erl
===>      Compiled clj_protocol.erl
===>      Compiled clj_murmur3.erl
===>      Compiled clj_module.erl
===>      Compiled clj_env.erl
===>      Compiled clj_emitter_pattern.erl
===>      Compiled clj_emitter.erl
===>      Compiled clj_compiler.erl
===>      Compiled clj_cache.erl
===>      Compiled clj_analyzer.erl
===> sh info:
        cwd: "d:/shaman/dev/clojerl"
        cmd: git log -n 1 --pretty=format:"%h
"

===>    opts: [{use_stdout,false},
                       return_on_error,
                       {cd,"d:/shaman/dev/clojerl"}]

===> Port Cmd: cmd /q /c git log -n 1 --pretty=format:"%h
"
Port Opts: [{cd,"d:/shaman/dev/clojerl"},
            exit_status,
            {line,16384},
            use_stdio,stderr_to_stdout,hide,eof]

===> sh info:
        cwd: "d:/shaman/dev/clojerl"
        cmd: git log --oneline --no-walk --tags --decorate

===>    opts: [{use_stdout,false},
                       return_on_error,
                       {cd,"d:/shaman/dev/clojerl"}]

===> Port Cmd: cmd /q /c git log --oneline --no-walk --tags --decorate
Port Opts: [{cd,"d:/shaman/dev/clojerl"},
            exit_status,
            {line,16384},
            use_stdio,stderr_to_stdout,hide,eof]

===> sh info:
        cwd: "d:/shaman/dev/clojerl"
        cmd: git describe --tags --abbrev=0

===>    opts: [{use_stdout,false},
                       return_on_error,
                       {cd,"d:/shaman/dev/clojerl"}]

===> Port Cmd: cmd /q /c git describe --tags --abbrev=0
Port Opts: [{cd,"d:/shaman/dev/clojerl"},
            exit_status,
            {line,16384},
            use_stdio,stderr_to_stdout,hide,eof]

===> sh info:
        cwd: "d:/shaman/dev/clojerl"
        cmd: git rev-list HEAD

===>    opts: [{use_stdout,false},
                       {cd,"d:/shaman/dev/clojerl"},
                       {debug_abort_on_error,
                           "Getting rev-list of git depedency failed in d:/shaman/dev/cloj
erl"}]

===> Port Cmd: cmd /q /c git rev-list HEAD
Port Opts: [{cd,"d:/shaman/dev/clojerl"},
            exit_status,
            {line,16384},
            use_stdio,stderr_to_stdout,hide,eof]

===> sh info:
        cwd: "d:/shaman/dev/clojerl"
        cmd: make EBIN=$REBAR_DEPS_DIR/clojerl/ebin clojure

===>    opts: [use_stdout,
                       {cd,"d:/shaman/dev/clojerl"},
                       {env,[{"REBAR_DEPS_DIR",
                              "d:/shaman/dev/clojerl/_build/default/lib"},
                             {"REBAR_BUILD_DIR",
                              "d:/shaman/dev/clojerl/_build/default"},
                             {"REBAR_ROOT_DIR","d:/shaman/dev/clojerl/."},
                             {"REBAR_CHECKOUTS_DIR",
                              "d:/shaman/dev/clojerl/_checkouts"},
                             {"REBAR_PLUGINS_DIR",
                              "d:/shaman/dev/clojerl/_build/default/plugins"},
                             {"REBAR_GLOBAL_CONFIG_DIR",
                              "c:/Users/qq563/.config/rebar3"},
                             {"REBAR_GLOBAL_CACHE_DIR",
                              "c:/Users/qq563/.cache/rebar3"},
                             {"REBAR_TEMPLATE_DIR",
                              "c:/Users/qq563/.config/rebar3/templates"},
                             {"REBAR_APP_DIRS",
                              "d:/shaman/dev/clojerl/_build/default/apps/*:d:/shaman/dev/c
lojerl/_build/default/lib/*:d:/shaman/dev/clojerl/_build/default/."},
                             {"REBAR_SRC_DIRS",[]},
                             {"ERLANG_ERTS_VER","9.0"},
                             {"ERLANG_ROOT_DIR","c:/Program Files/erl9.0"},
                             {"ERLANG_LIB_DIR_erl_interface",
                              "c:/Program Files/erl9.0/lib/erl_interface-3.10"},
                             {"ERLANG_LIB_VER_erl_interface","3.10"},
                             {"ERL","c:/Program Files/erl9.0/bin/erl"},
                             {"ERLC","c:/Program Files/erl9.0/bin/erlc"},
                             {"ERLANG_ARCH","64"},
                             {"ERLANG_TARGET","20.0-win32-64"}]},
                       {abort_on_error,"Hook for compile failed!\n"}]

===> Port Cmd: cmd /q /c make EBIN=/clojerl/ebin clojure
Port Opts: [{cd,"d:/shaman/dev/clojerl"},
            {env,[{"REBAR_DEPS_DIR",
                   "d:/shaman/dev/clojerl/_build/default/lib"},
                  {"REBAR_BUILD_DIR","d:/shaman/dev/clojerl/_build/default"},
                  {"REBAR_ROOT_DIR","d:/shaman/dev/clojerl/."},
                  {"REBAR_CHECKOUTS_DIR","d:/shaman/dev/clojerl/_checkouts"},
                  {"REBAR_PLUGINS_DIR",
                   "d:/shaman/dev/clojerl/_build/default/plugins"},
                  {"REBAR_GLOBAL_CONFIG_DIR","c:/Users/qq563/.config/rebar3"},
                  {"REBAR_GLOBAL_CACHE_DIR","c:/Users/qq563/.cache/rebar3"},
                  {"REBAR_TEMPLATE_DIR",
                   "c:/Users/qq563/.config/rebar3/templates"},
                  {"REBAR_APP_DIRS",
                   "d:/shaman/dev/clojerl/_build/default/apps/*:d:/shaman/dev/clojerl/_bui
ld/default/lib/*:d:/shaman/dev/clojerl/_build/default/."},
                  {"REBAR_SRC_DIRS",[]},
                  {"ERLANG_ERTS_VER","9.0"},
                  {"ERLANG_ROOT_DIR","c:/Program Files/erl9.0"},
                  {"ERLANG_LIB_DIR_erl_interface",
                   "c:/Program Files/erl9.0/lib/erl_interface-3.10"},
                  {"ERLANG_LIB_VER_erl_interface","3.10"},
                  {"ERL","c:/Program Files/erl9.0/bin/erl"},
                  {"ERLC","c:/Program Files/erl9.0/bin/erlc"},
                  {"ERLANG_ARCH","64"},
                  {"ERLANG_TARGET","20.0-win32-64"}]},
            exit_status,
            {line,16384},
            use_stdio,stderr_to_stdout,hide,eof]

Compiling clojure.core.server...escript: exception error: undefined function clojerl:start
/0
  in function  erl_eval:do_apply/6 (erl_eval.erl, line 670)
  in call from erl_eval:expr/5 (erl_eval.erl, line 438)
  in call from escript:eval_exprs/5 (escript.erl, line 873)
  in call from erl_eval:local_func/6 (erl_eval.erl, line 557)
  in call from escript:interpret/4 (escript.erl, line 789)
  in call from escript:start/1 (escript.erl, line 277)
  in call from init:start_em/1 (init.erl, line 1083)
make[1]: *** [Makefile:97: /clojerl/ebin/clojure.core.server.beam] Error 127
===> Hook for compile failed!

make: *** [Makefile:17: compile] Error 1
`

$ git pull
remote: Counting objects: 3, done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 3 (delta 1), reused 3 (delta 1), pack-reused 0
warning: unable to unlink .git/objects/8f/tmp_obj_En8jy4: Device or resource busy
Unpacking objects: 100% (3/3), done.
From https://github.com/jfacorro/clojerl
 + 6a46956...8fc704c jfacorro.387.build.on.windows -> origin/jfacorro.387.build.on.windows
  (forced update)
Already up-to-date.

qq563@DESKTOP-GWW  ~/dev/clojerl
$ make clean; DEBUG=1 make | tee rebar3.debug
[0;32m===> Cleaning out clojerl...
[0m===> Evaluating config script "rebar.config.script"
===> Expanded command sequence to be run: [{default,app_discovery},
                                                  {default,install_deps},
                                                  {default,lock},
                                                  {default,compile}]
===> Evaluating config script "d:/shaman/dev/clojerl/rebar.config.script"
===> Evaluating config script "d:/shaman/dev/clojerl/rebar.config.script"
===> Verifying dependencies...
===> run_hooks("d:/shaman/dev/clojerl", pre_hooks, compile) -> no hooks defined

===> sh info:
        cwd: "d:/shaman/dev/clojerl"
        cmd: cmd /c mklink /j "d:\\shaman\\dev\\clojerl\\_build\\default\\lib\\clojerl\\in
clude" "d:\\shaman\\dev\\clojerl\\include"

===>    opts: [{use_stdout,false},return_on_error]

===> Port Cmd: cmd /q /c cmd /c mklink /j "d:\\shaman\\dev\\clojerl\\_build\\default\\lib\
\clojerl\\include" "d:\\shaman\\dev\\clojerl\\include"
Port Opts: [exit_status,{line,16384},use_stdio,stderr_to_stdout,hide,eof]

===> sh info:
        cwd: "d:/shaman/dev/clojerl"
        cmd: cmd /c mklink /j "d:\\shaman\\dev\\clojerl\\_build\\default\\lib\\clojerl\\sr
c" "d:\\shaman\\dev\\clojerl\\src"

===>    opts: [{use_stdout,false},return_on_error]

===> Port Cmd: cmd /q /c cmd /c mklink /j "d:\\shaman\\dev\\clojerl\\_build\\default\\lib\
\clojerl\\src" "d:\\shaman\\dev\\clojerl\\src"
Port Opts: [exit_status,{line,16384},use_stdio,stderr_to_stdout,hide,eof]

===> Compiling clojerl
===> erlopts [debug_info,debug_info]
===> files to compile ["d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/rbdict.er
l",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IType.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IStringable.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IStack.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.ISet.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.ISequential.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.ISeqable.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.ISeq.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IReduce.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IRecord.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.INamed.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IMeta.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IMap.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.ILookup.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IIndexed.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IHash.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IFn.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IExceptionInfo.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IError.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IEquiv.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IDeref.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.ICounted.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IColl.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IChunkedSeq.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IChunk.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/protocols/clojerl.IAssociative.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.Vector.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.Vector.ChunkedSeq.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.TupleMap.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.TupleChunk.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.SortedSet.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.SortedMap.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.Set.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.Range.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.Map.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.List.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.LazySeq.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.Cons.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/collections/clojerl.ChunkedCons.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Var.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Symbol.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.String.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Reduced.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.reader.TaggedLiteral.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.reader.ReaderConditional.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Nil.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Namespace.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.MultiFn.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Keyword.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Integer.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Float.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.ExceptionInfo.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Default.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Boolean.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.BitString.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/la
ng/clojerl.Atom.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.util.UUID.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.util.Regex.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.util.Date.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.Type.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.Tuple.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.Reference.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.Process.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.Port.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.Map.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.List.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.io.StringWriter.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.io.StringReader.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.io.PushbackReader.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.io.IWriter.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.io.IReader.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.io.IPushbackReader.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.io.File.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.io.Closeable.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/er
lang/erlang.Fn.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
ojerl_sup.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
ojerl.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_utils.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_scope.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_rt.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_reader.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_protocol.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_murmur3.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_module.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_env.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_emitter_pattern.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_emitter.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_compiler.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_cache.erl",
                              "d:/shaman/dev/clojerl/_build/default/lib/clojerl/src/erl/cl
j_analyzer.erl"]
===>      Compiled erlang.io.IPushbackReader.erl
===>      Compiled clojerl.ISeq.erl
===>      Compiled clojerl.ISet.erl
===>      Compiled erlang.io.IWriter.erl
===>      Compiled erlang.io.IReader.erl
===>      Compiled clojerl.IExceptionInfo.erl
===>      Compiled clojerl.IColl.erl
===>      Compiled clojerl.IHash.erl
===>      Compiled clojerl.IMap.erl
===>      Compiled clojerl.IIndexed.erl
===>      Compiled clojerl.IChunkedSeq.erl
===>      Compiled clojerl.IDeref.erl
===>      Compiled clojerl.INamed.erl
===>      Compiled clojerl.IChunk.erl
===>      Compiled clojerl.ISequential.erl
===>      Compiled clojerl.IAssociative.erl
===>      Compiled clojerl.IStringable.erl
===>      Compiled clojerl.IError.erl
===>      Compiled clojerl.IFn.erl
===>      Compiled clojerl.IMeta.erl
===>      Compiled erlang.io.Closeable.erl
===>      Compiled clojerl.IReduce.erl
===>      Compiled clojerl.IEquiv.erl
===>      Compiled clojerl.ISeqable.erl
===>      Compiled clojerl.ILookup.erl
===>      Compiled clojerl.IStack.erl
===>      Compiled clojerl.ICounted.erl
===>      Compiled rbdict.erl
===>      Compiled clojerl.IType.erl
===>      Compiled clojerl.IRecord.erl
===>      Compiled clojerl.Vector.erl
===>      Compiled clojerl.Vector.ChunkedSeq.erl
===>      Compiled clojerl.TupleMap.erl
===>      Compiled clojerl.TupleChunk.erl
===>      Compiled clojerl.SortedSet.erl
===>      Compiled clojerl.SortedMap.erl
===>      Compiled clojerl.Set.erl
===>      Compiled clojerl.Range.erl
===>      Compiled clojerl.Map.erl
===>      Compiled clojerl.List.erl
===>      Compiled clojerl.LazySeq.erl
===>      Compiled clojerl.Cons.erl
===>      Compiled clojerl.ChunkedCons.erl
===>      Compiled clojerl.Var.erl
===>      Compiled clojerl.Symbol.erl
===>      Compiled clojerl.String.erl
===>      Compiled clojerl.Reduced.erl
===>      Compiled clojerl.reader.TaggedLiteral.erl
===>      Compiled clojerl.reader.ReaderConditional.erl
===>      Compiled clojerl.Nil.erl
===>      Compiled clojerl.Namespace.erl
===>      Compiled clojerl.MultiFn.erl
===>      Compiled clojerl.Keyword.erl
===>      Compiled clojerl.Integer.erl
===>      Compiled clojerl.Float.erl
===>      Compiled clojerl.ExceptionInfo.erl
===>      Compiled clojerl.Default.erl
===>      Compiled clojerl.Boolean.erl
===>      Compiled clojerl.BitString.erl
===>      Compiled clojerl.Atom.erl
===>      Compiled erlang.util.UUID.erl
===>      Compiled erlang.util.Regex.erl
===>      Compiled erlang.util.Date.erl
===>      Compiled erlang.Type.erl
===>      Compiled erlang.Tuple.erl
===>      Compiled erlang.Reference.erl
===>      Compiled erlang.Process.erl
===>      Compiled erlang.Port.erl
===>      Compiled erlang.Map.erl
===>      Compiled erlang.List.erl
===>      Compiled erlang.io.StringWriter.erl
===>      Compiled erlang.io.StringReader.erl
===>      Compiled erlang.io.PushbackReader.erl
===>      Compiled erlang.io.File.erl
===>      Compiled erlang.Fn.erl
===>      Compiled clojerl_sup.erl
===>      Compiled clojerl.erl
===>      Compiled clj_utils.erl
===>      Compiled clj_scope.erl
===>      Compiled clj_rt.erl
===>      Compiled clj_reader.erl
===>      Compiled clj_protocol.erl
===>      Compiled clj_murmur3.erl
===>      Compiled clj_module.erl
===>      Compiled clj_env.erl
===>      Compiled clj_emitter_pattern.erl
===>      Compiled clj_emitter.erl
===>      Compiled clj_compiler.erl
===>      Compiled clj_cache.erl
===>      Compiled clj_analyzer.erl
===> sh info:
        cwd: "d:/shaman/dev/clojerl"
        cmd: git log -n 1 --pretty=format:"%h
"

===>    opts: [{use_stdout,false},
                       return_on_error,
                       {cd,"d:/shaman/dev/clojerl"}]

===> Port Cmd: cmd /q /c git log -n 1 --pretty=format:"%h
"
Port Opts: [{cd,"d:/shaman/dev/clojerl"},
            exit_status,
            {line,16384},
            use_stdio,stderr_to_stdout,hide,eof]

===> sh info:
        cwd: "d:/shaman/dev/clojerl"
        cmd: git log --oneline --no-walk --tags --decorate

===>    opts: [{use_stdout,false},
                       return_on_error,
                       {cd,"d:/shaman/dev/clojerl"}]

===> Port Cmd: cmd /q /c git log --oneline --no-walk --tags --decorate
Port Opts: [{cd,"d:/shaman/dev/clojerl"},
            exit_status,
            {line,16384},
            use_stdio,stderr_to_stdout,hide,eof]

===> sh info:
        cwd: "d:/shaman/dev/clojerl"
        cmd: git describe --tags --abbrev=0

===>    opts: [{use_stdout,false},
                       return_on_error,
                       {cd,"d:/shaman/dev/clojerl"}]

===> Port Cmd: cmd /q /c git describe --tags --abbrev=0
Port Opts: [{cd,"d:/shaman/dev/clojerl"},
            exit_status,
            {line,16384},
            use_stdio,stderr_to_stdout,hide,eof]

===> sh info:
        cwd: "d:/shaman/dev/clojerl"
        cmd: git rev-list HEAD

===>    opts: [{use_stdout,false},
                       {cd,"d:/shaman/dev/clojerl"},
                       {debug_abort_on_error,
                           "Getting rev-list of git depedency failed in d:/shaman/dev/cloj
erl"}]

===> Port Cmd: cmd /q /c git rev-list HEAD
Port Opts: [{cd,"d:/shaman/dev/clojerl"},
            exit_status,
            {line,16384},
            use_stdio,stderr_to_stdout,hide,eof]

===> sh info:
        cwd: "d:/shaman/dev/clojerl"
        cmd: make EBIN=$REBAR_DEPS_DIR/clojerl/ebin clojure

===>    opts: [use_stdout,
                       {cd,"d:/shaman/dev/clojerl"},
                       {env,[{"REBAR_DEPS_DIR",
                              "d:/shaman/dev/clojerl/_build/default/lib"},
                             {"REBAR_BUILD_DIR",
                              "d:/shaman/dev/clojerl/_build/default"},
                             {"REBAR_ROOT_DIR","d:/shaman/dev/clojerl/."},
                             {"REBAR_CHECKOUTS_DIR",
                              "d:/shaman/dev/clojerl/_checkouts"},
                             {"REBAR_PLUGINS_DIR",
                              "d:/shaman/dev/clojerl/_build/default/plugins"},
                             {"REBAR_GLOBAL_CONFIG_DIR",
                              "c:/Users/qq563/.config/rebar3"},
                             {"REBAR_GLOBAL_CACHE_DIR",
                              "c:/Users/qq563/.cache/rebar3"},
                             {"REBAR_TEMPLATE_DIR",
                              "c:/Users/qq563/.config/rebar3/templates"},
                             {"REBAR_APP_DIRS",
                              "d:/shaman/dev/clojerl/_build/default/apps/*:d:/shaman/dev/c
lojerl/_build/default/lib/*:d:/shaman/dev/clojerl/_build/default/."},
                             {"REBAR_SRC_DIRS",[]},
                             {"ERLANG_ERTS_VER","9.0"},
                             {"ERLANG_ROOT_DIR","c:/Program Files/erl9.0"},
                             {"ERLANG_LIB_DIR_erl_interface",
                              "c:/Program Files/erl9.0/lib/erl_interface-3.10"},
                             {"ERLANG_LIB_VER_erl_interface","3.10"},
                             {"ERL","c:/Program Files/erl9.0/bin/erl"},
                             {"ERLC","c:/Program Files/erl9.0/bin/erlc"},
                             {"ERLANG_ARCH","64"},
                             {"ERLANG_TARGET","20.0-win32-64"}]},
                       {abort_on_error,"Hook for compile failed!\n"}]

===> Port Cmd: cmd /q /c make EBIN=/clojerl/ebin clojure
Port Opts: [{cd,"d:/shaman/dev/clojerl"},
            {env,[{"REBAR_DEPS_DIR",
                   "d:/shaman/dev/clojerl/_build/default/lib"},
                  {"REBAR_BUILD_DIR","d:/shaman/dev/clojerl/_build/default"},
                  {"REBAR_ROOT_DIR","d:/shaman/dev/clojerl/."},
                  {"REBAR_CHECKOUTS_DIR","d:/shaman/dev/clojerl/_checkouts"},
                  {"REBAR_PLUGINS_DIR",
                   "d:/shaman/dev/clojerl/_build/default/plugins"},
                  {"REBAR_GLOBAL_CONFIG_DIR","c:/Users/qq563/.config/rebar3"},
                  {"REBAR_GLOBAL_CACHE_DIR","c:/Users/qq563/.cache/rebar3"},
                  {"REBAR_TEMPLATE_DIR",
                   "c:/Users/qq563/.config/rebar3/templates"},
                  {"REBAR_APP_DIRS",
                   "d:/shaman/dev/clojerl/_build/default/apps/*:d:/shaman/dev/clojerl/_bui
ld/default/lib/*:d:/shaman/dev/clojerl/_build/default/."},
                  {"REBAR_SRC_DIRS",[]},
                  {"ERLANG_ERTS_VER","9.0"},
                  {"ERLANG_ROOT_DIR","c:/Program Files/erl9.0"},
                  {"ERLANG_LIB_DIR_erl_interface",
                   "c:/Program Files/erl9.0/lib/erl_interface-3.10"},
                  {"ERLANG_LIB_VER_erl_interface","3.10"},
                  {"ERL","c:/Program Files/erl9.0/bin/erl"},
                  {"ERLC","c:/Program Files/erl9.0/bin/erlc"},
                  {"ERLANG_ARCH","64"},
                  {"ERLANG_TARGET","20.0-win32-64"}]},
            exit_status,
            {line,16384},
            use_stdio,stderr_to_stdout,hide,eof]

Compiling clojure.core.server...escript: exception error: undefined function clojerl:start
/0
  in function  erl_eval:do_apply/6 (erl_eval.erl, line 670)
  in call from erl_eval:expr/5 (erl_eval.erl, line 438)
  in call from escript:eval_exprs/5 (escript.erl, line 873)
  in call from erl_eval:local_func/6 (erl_eval.erl, line 557)
  in call from escript:interpret/4 (escript.erl, line 789)
  in call from escript:start/1 (escript.erl, line 277)
  in call from init:start_em/1 (init.erl, line 1083)
make[1]: *** [Makefile:98: /clojerl/ebin/clojure.core.server.beam] Error 127
===> Hook for compile failed!

make: *** [Makefile:18: compile] Error 1

qq563@DESKTOP-GWW  ~/dev/clojerl
$ git pull
Already up-to-date.

--- a/.appveyor.yml
+++ b/.appveyor.yml
@@ -1,0 +1,10 @@
+version: "{build}"
+build: off
+environment:
+  matrix:
+  - erlang_vsn: 20.0
+install:
+  - ps: choco install erlang --version $env:erlang_vsn
+  - ps: wget https://s3.amazonaws.com/rebar3/rebar3 -OutFile C:\msys64\usr\bin\rebar3
+test_script:
+  - cmd: C:\msys64\usr\bin\bash.exe --login -c "export PATH=/mingw64/bin:$PATH; cd /c/projects/clojerl; make ci"

--- a/.gitignore
+++ b/.gitignore
@@ -6,6 +6,7 @@ clojerl.d
 callgrind.*
 *.so
 *.o
+*.dll
 c_src/*.mk
 *.plt
 *.out*

--- a/.travis.yml
+++ b/.travis.yml
@@ -8,6 +8,6 @@ install:
   - chmod +x rebar3
   - export PATH=$PATH:$PWD
 script:
-  - make travis-ci
+  - make ci
 notifications:
   email: false

--- a/Makefile
+++ b/Makefile
@@ -1,9 +1,14 @@
 .SILENT:
 
 REBAR3    := rebar3
+RLWRAP    := $(shell type rlwrap &> /dev/null && echo rlwrap || echo)
 V         := @
 
 EBIN      ?= ${CURDIR}/ebin
+ifeq (${NO_CLOJURE},)
+	EBIN = ${REBAR_DEPS_DIR}/clojerl/ebin
+endif
+
 ERL_SRC   := ${CURDIR}/src/erl
 INCLUDE   := ${CURDIR}/include
 ERL_OPTS  := +debug_info
@@ -32,12 +37,12 @@ clean:
 	${V} rm -rf ${EBIN}
 	${V} rm -rf priv/*.so c_src/*.o
 
-travis-ci: test dialyzer
+ci: test dialyzer
 
 repl: SHELL_OPTS = -sname clojerl-repl -setcookie clojerl -s clojerl
 repl: SHELL_OPTS += -eval "'clojure.main':main([<<\"-r\">>])." -s clojerl start -noshell +pc unicode
 repl: compile
-	${V} rlwrap erl -pa `rebar3 path --ebin` ${CODE_PATH} ${SHELL_OPTS}
+	${V} ${RLWRAP} erl -pa `rebar3 path --ebin` ${CODE_PATH} ${SHELL_OPTS}
 
 # ------------------------------------------------------------------------------
 # Clojure compilation

--- a/c_src/Makefile
+++ b/c_src/Makefile
@@ -11,10 +11,17 @@ ERL_INTERFACE_INCLUDE_DIR ?= $(shell erl -noshell -s init stop -eval "io:format(
 ERL_INTERFACE_LIB_DIR ?= $(shell erl -noshell -s init stop -eval "io:format(\"~s\", [code:lib_dir(erl_interface, lib)]).")
 
 C_SRC_DIR = $(CURDIR)
-C_SRC_OUTPUT ?= $(CURDIR)/../priv/$(PROJECT).so
+C_SRC_OUTPUT ?= $(CURDIR)/../priv/$(PROJECT)
+
+ifeq ($(OS), Windows_NT)
+	C_SRC_OUTPUT_SHARED_EXTENSION ?= .dll
+else
+	C_SRC_OUTPUT_SHARED_EXTENSION ?= .so
+endif
+
+C_SRC_OUTPUT_FILE = $(C_SRC_OUTPUT)$(C_SRC_OUTPUT_SHARED_EXTENSION)
 
 # System type and C compiler/flags.
-
 UNAME_SYS := $(shell uname -s)
 ifeq ($(UNAME_SYS), Darwin)
 	CC ?= cc
@@ -29,12 +36,22 @@ else ifeq ($(UNAME_SYS), Linux)
 	CC ?= gcc
 	CFLAGS ?= -O3 -std=c99 -finline-functions -Wall -Wmissing-prototypes
 	CXXFLAGS ?= -O3 -finline-functions -Wall
+else ifeq ($(OS), Windows_NT)
+	CC = /mingw64/bin/gcc
+	export CC
+	CFLAGS ?= -O3 -std=c99 -finline-functions -Wall -Wmissing-prototypes
+	CXXFLAGS ?= -O3 -finline-functions -Wall
 endif
 
-CFLAGS += -fPIC -I $(ERTS_INCLUDE_DIR) -I $(ERL_INTERFACE_INCLUDE_DIR)
-CXXFLAGS += -fPIC -I $(ERTS_INCLUDE_DIR) -I $(ERL_INTERFACE_INCLUDE_DIR)
+ifneq ($(OS),Windows_NT)
+	CFLAGS += -fPIC
+	CXXFLAGS += -fPIC
+endif
 
-LDLIBS += -L $(ERL_INTERFACE_LIB_DIR) -lerl_interface -lei
+CFLAGS += -I"$(ERTS_INCLUDE_DIR)" -I"$(ERL_INTERFACE_INCLUDE_DIR)"
+CXXFLAGS += -I"$(ERTS_INCLUDE_DIR)" -I"$(ERL_INTERFACE_INCLUDE_DIR)"
+
+LDLIBS += -L "$(ERL_INTERFACE_LIB_DIR)" -lerl_interface -lei
 LDFLAGS += -shared
 
 # Verbosity.
@@ -54,9 +71,9 @@ OBJECTS = $(addsuffix .o, $(basename $(SOURCES)))
 COMPILE_C = $(c_verbose) $(CC) $(CFLAGS) $(CPPFLAGS) -c
 COMPILE_CPP = $(cpp_verbose) $(CXX) $(CXXFLAGS) $(CPPFLAGS) -c
 
-$(C_SRC_OUTPUT): $(OBJECTS)
+$(C_SRC_OUTPUT_FILE): $(OBJECTS)
 	@mkdir -p $(BASEDIR)/priv/
-	$(link_verbose) $(CC) $(OBJECTS) $(LDFLAGS) $(LDLIBS) -o $(C_SRC_OUTPUT)
+	$(link_verbose) $(CC) $(OBJECTS) $(LDFLAGS) $(LDLIBS) -o $(C_SRC_OUTPUT_FILE)
 
 %.o: %.c
 	$(COMPILE_C) $(OUTPUT_OPTION) $<

--- a/rebar.config
+++ b/rebar.config
@@ -16,10 +16,9 @@
            ]}
   ]}.
 
-{pre_hooks,  [ {"(linux|darwin|solaris)", compile, "make -C c_src PROJECT=clojerl"}
-             , {"(linux|darwin|solaris)", clean,   "make -C c_src clean PROJECT=clojerl"}
+{pre_hooks,  [ {"(linux|darwin|solaris|win32|win64)", compile, "make -C c_src PROJECT=clojerl"}
+             , {"(linux|darwin|solaris|win32|win64)", clean,   "make -C c_src clean PROJECT=clojerl"}
              ]}.
-{artifacts,  ["priv/clojerl.so"]}.
 
 {cover_enabled, true}.
 {shell, [{apps, [clojerl, sync]}]}.

--- a/rebar.config.script
+++ b/rebar.config.script
@@ -1,6 +1,6 @@
 case os:getenv("NO_CLOJURE") of
   false ->
-    [ {post_hooks, [{compile, "make EBIN=$REBAR_DEPS_DIR/clojerl/ebin clojure"}]}
+    [ {post_hooks, [{compile, "make clojure"}]}
       | CONFIG
     ];
   _ ->

